contracts/HolographBridge.sol

```diff
@@ -154,251 +153,209 @@ contract HolographBridge is Admin, Initializable, IHolographBridge {
       (address, address, address, address, address)
     );
     assembly {
-      sstore(0x5705f5753aa4f617eef2cae1dada3d3355e9387b04d19191f09b545e684ca50d, origin())
+      sstore(_adminSlot, origin())
 
-      sstore(0x7eefc8e705e14d34b5d1d6c3ea7f4e20cecb5956b182bac952a455d9372b87e2, factory)
-      sstore(0x1eee493315beeac80829afd0aaa340f3821cabe68571a2743478e81638a3d94d, holograph)
-      sstore(0x23e584d4fb363739321c1e56c9bcdc29517a4c57065f8502226c995fd15b2472, interfaces)
-      sstore(0x7bef7d8d97f57f9aa64de319c8598b5cdc7c3d2715fc02428415a98281ca6bdc, operator)
-      sstore(0x460c4059d72b144253e5fc4e2aacbae2bcd6362c67862cd58ecbab0e7b10c349, registry)
+      sstore(_factorySlot, factory)
+      sstore(_holographSlot, holograph)
+      sstore(_interfacesSlot, interfaces)
+      sstore(_operatorSlot, operator)
+      sstore(_registrySlot, registry)
     }
     _setInitialized();
     return IInitializable.init.selector;
   }
 
-  function executeJob(bytes calldata _payload) external onlyOperator {
-    assembly {
-      calldatacopy(0, _payload.offset, _payload.length)
-      let result := callcode(gas(), address(), callvalue(), 0, _payload.length, 0, 0)
-      if eq(result, 0) {
-        returndatacopy(0, 0, returndatasize())
-        revert(0, returndatasize())
-      }
-    }
-  }
-
-  function erc721in(
+  function bridgeInRequest(
+    uint256, /* nonce*/
     uint32 fromChain,
-    address collection,
-    address from,
-    address to,
-    uint256 tokenId,
+    address holographableContract,
+    address hToken,
+    address hTokenRecipient,
+    uint256 hTokenValue,
     bytes calldata data
-  ) external onlyBridge {
-    require(_registry().isHolographedContract(collection), "HOLOGRAPH: not holographed");
+  ) external onlyOperator {
     require(
-      ERC721Holograph(collection).holographBridgeIn(fromChain, from, to, tokenId, data) ==
-        ERC721Holograph.holographBridgeIn.selector,
-      "HOLOGRAPH: bridge in failed"
+      _registry().isHolographedContract(holographableContract) || address(_factory()) == holographableContract,
+      "HOLOGRAPH: not holographed"
     );
+    bytes4 selector = HolographableEnforcer(holographableContract).bridgeIn(fromChain, data);
+    require(selector == HolographableEnforcer.bridgeIn.selector, "HOLOGRAPH: bridge in failed");
+    if (hTokenValue > 0) {
+      // provide operator with hToken value for executing bridge job
+      require(
```

Comment:
Does this actually send out the value, or just check it?

Reply:
This is here for now to keep the code from reverting while HLG logic is being implemented.

Reply:
gotcha